core/src/main/java/io/questdb/griffin/SqlCompiler.java

```diff
@@ -1581,6 +1650,178 @@ private void validateTableModelAndCreateTypeCast(
             throw SqlException.position(0).put("timestamp is not defined");
         }
     }
+    
+    private CompiledQuery sqlBackup(SqlExecutionContext executionContext) throws SqlException {
+        if (null == configuration.getBackupRoot()) {
+            throw CairoException.instance(0).put("Backup is disabled, no backup root directory is configured in the server configuration ['cairo.sql.backup.root' property]");
+        }
+
+        final CharSequence tok = SqlUtil.fetchNext(lexer);
+        if (null != tok) {
+            if (Chars.equalsLowerCaseAscii(tok, "table")) {
+                return sqlTableBackup(executionContext);
+            }
+            if (Chars.equalsLowerCaseAscii(tok, "database")) {
+                return sqlDatabaseBackup(executionContext);
+            }
+        }
+
+        throw SqlException.position(lexer.lastTokenPosition()).put(" expected 'table' or 'database'");
+    }
+
+    private final FindVisitor sqlDatabaseBackupOnFind = (file, type) -> {
+        nativeLPSZ.of(file);
+        if (type == Files.DT_DIR && nativeLPSZ.charAt(0) != '.') {
+            backupTable(nativeLPSZ, currentExecutionContext);
+        }
+    };
+
+    private CompiledQuery sqlDatabaseBackup(SqlExecutionContext executionContext) throws SqlException {
+        currentExecutionContext = executionContext;
+        try {
+            setupBackupRenamePath();
+            ff.iterateDir(path.of(configuration.getRoot()).$(), sqlDatabaseBackupOnFind);
```

Comment:
I feel this loop should be robust to directory creating corrupt tables, for example any table on which `backupTable`  might fail.

This would make interface relatively consistent. We should be able to say that backup database will backup everything than can be backed up and will log tables it could not backup.